docs/how-tos/sensor-module.md

```diff
@@ -0,0 +1,414 @@
+        return
+
+    # Handles attribute reconfiguration
+    def reconfigure(
+      self, config: ComponentConfig,
+      dependencies: Mapping[ResourceName, ResourceBase]
+      ):
+        self.latitude = float(
+          config.attributes.fields["latitude"].number_value)
+        self.longitude = float(
+          config.attributes.fields["longitude"].number_value)
+        return
+
+    async def get_readings(
+        self, *, extra: Optional[Mapping[str, Any]] = None,
+        timeout: Optional[float] = None, **kwargs
+    ) -> Mapping[str, SensorReading]:
+
+        # Set up the Open-Meteo API client with cache and retry on error
+        cache_session = requests_cache.CachedSession(
+          '.cache', expire_after=3600)
+        retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+        openmeteo = openmeteo_requests.Client(session=retry_session)
+
+        # The order of variables in hourly or daily is
+        # important to assign them correctly below
+        url = "https://air-quality-api.open-meteo.com/v1/air-quality"
+        params = {
+            "latitude": self.latitude,
+            "longitude": self.longitude,
+            "current": ["pm10", "pm2_5"],
+            "timezone": "America/Los_Angeles"
+        }
+        responses = openmeteo.weather_api(url, params=params)
+
+        # Process location
+        response = responses[0]
+
+        # Current values. The order of variables needs
+        # to be the same as requested.
+        current = response.Current()
+        current_pm10 = current.Variables(0).Value()
+        current_pm2_5 = current.Variables(1).Value()
+
+        LOGGER.info(current_pm2_5)
+
+        # Return a dictionary of the readings
+        return {
+            "pm2_5": current_pm2_5,
+            "pm10": current_pm10
+        }
+```
+
+{{< /expand >}}
+
+For more examples, see the [existing sensor models](/components/sensor/#available-models).
+
+## Edit requirements.txt
+
+Update the generated <file>requirements.txt</file> file to include any packages that must be installed for the module to run.
+Depending on your use case, you may not need to add anything here beyond `viam-sdk` which is auto-populated.
+
+## Make the module executable
+
+You need an executable file so that `viam-server` can run your module.
+The module generator already created the <file>run.sh</file> "entrypoint" file for you, so all you need to do is make this file executable by running the following command with your correct file path:
+
+```sh {id="terminal-prompt" class="command-line" data-prompt="$"}
+sudo chmod +x <your-file-path-to>/run.sh
+```
+
+## Test your module locally
+
+It's a good idea to test your module locally before uploading it to the [Viam registry](https://app.viam.com/registry):
+
+1. Follow [the local module testing instructions](/how-tos/create-module/#test-your-module-locally) to configure your local module and then the associated model on your machine.
```

Comment:
I think those steps are so small state you can include them directly, maybe in slightly shortened form